RFS/src/test/java/com/rfs/integration/SnapshotStateTest.java

```diff
@@ -0,0 +1,248 @@
+    private static String clusterUrl;
+
+    @SuppressWarnings("resource")
+    @BeforeEach
+    public void setUp() throws Exception {
+        IOUtils.rm(Path.of(SNAPSHOT_DIR.getAbsolutePath()));
+        SNAPSHOT_DIR.mkdir();
+
+        clusterContainer = new GenericContainer<>(DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch:7.10.2"))
+                .withExposedPorts(9200, 9300)
+                .withEnv("discovery.type", "single-node")
+                .withEnv("path.repo", "/usr/share/elasticsearch/snapshots")
+                .withFileSystemBind(SNAPSHOT_DIR.getName(), "/usr/share/elasticsearch/snapshots", BindMode.READ_WRITE)
+                .withLogConsumer(new Slf4jLogConsumer(logger))
+                .waitingFor(Wait.forHttp("/").forPort(9200).forStatusCode(200).withStartupTimeout(Duration.ofMinutes(1)));
+
+        clusterContainer.start();
+
+        final var address = clusterContainer.getHost();
+        final var port = clusterContainer.getMappedPort(9200);
+        httpClient = HttpClients.createDefault();
+        clusterUrl = "http://" + address + ":" + port;
+
+        // Create snapshot repository
+        final var repositoryJson = "{\n" +
+        "  \"type\": \"fs\",\n" +
+        "  \"settings\": {\n" +
+        "    \"location\": \"/usr/share/elasticsearch/snapshots\",\n" +
+        "    \"compress\": false\n" +
+        "  }\n" +
+        "}";
+
+        final var createRepoRequest = new HttpPut(clusterUrl + "/_snapshot/test-repo");
+        createRepoRequest.setEntity(new StringEntity(repositoryJson));
+        createRepoRequest.setHeader("Content-Type", "application/json");
+
+        try (var response = httpClient.execute(createRepoRequest)) {
+            assertThat(response.getStatusLine().getStatusCode(), equalTo(200));
+        }
+    }
+
+    @AfterEach
+    public void tearDown() throws IOException {
+        System.out.println("Container Logs:\n" + clusterContainer.getLogs());
+
+        httpClient.close();
+        clusterContainer.stop();
+        IOUtils.rm(Path.of(SNAPSHOT_DIR.getAbsolutePath()));
+    }
+
+    public void createDocument(final String index, final String docId, final String body) throws IOException {
+        var indexDocumentRequset = new HttpPut(clusterUrl + "/" + index + "/_doc/" + docId);
+        indexDocumentRequset.setEntity(new StringEntity(body));
+        indexDocumentRequset.setHeader("Content-Type", "application/json");
+
+        try (var response = httpClient.execute(indexDocumentRequset)) {
+            assertThat(response.getStatusLine().getStatusCode(), equalTo(201));
+        }
+    }
+
+    public void takeSnapshot(final String snapshotName, final String indexPattern) throws IOException {
+        // Create snapshot
+        final var snapshotJson = "{\n" +
+                "  \"indices\": \"" + indexPattern + "\",\n" +
+                "  \"ignore_unavailable\": true,\n" +
+                "  \"include_global_state\": false\n" +
```

Comment:
We need to include the global state in order to perform metadata migrations (e.g. Templates).  Putting false here will cause that phase of RFS to fail because a file we're expecting to be the snapshot won't even be present.

Reply:
I've updated this to true, but the test passed.  We can revisit more as we 'hook up' all the pieces